### PR TITLE
fix(FEC-8376): the menu ui doesn't show the right caption after selection

### PR DESCRIPTION
**Pull request by odedhutzler**
updating the indexes only after there are tracks in the player.
updating the indexes only after there are tracks in the player and also notifying the ui on the indexes change.

### Description of the Changes

Please add a detailed description of the change, weather it's an enhancement or a bugfix.
If the PR is related to an open issue please link to it.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -1517,7 +1517,10 @@ export default class Player extends FakeEventTarget {
           const videoElement = this.getVideoElement();
           return videoElement ? Array.from(videoElement.textTracks).findIndex(track => track ? track.language === textTrack.language : false) : -1;
         };
-        this._getTracksByType(TrackType.TEXT).forEach(track => track.index = getNativeLanguageTrackIndex(track));
+        this.ready().then(() => {
```
Comment (OrenMe) [inline src/player.js]: If we already need to wait for ready event to map the index than it probably means that we don’t need to be handled here at all and move this to the ready promise chain just before we emit the regular TEXT_TRACK_CHANGED

Comment (odedhutzler) [reply src/player.js]: @OrenMe might there be a case when the ready promise will resolved before we add native tracks?

Comment (OrenMe) [reply src/player.js]: External native tracks are controlled by you and not by 3rd party lib that parses and downloads a manifest so this is something you have full control over. AFAICT no need for guessing work here

Comment (OrenMe) [reply src/player.js]: @odedhutzler what is the status of this?

**Pull request closed without merging**